src/libstore/derivations.cc

```diff
@@ -756,8 +756,13 @@ std::optional<BasicDerivation> Derivation::tryResolveUncached(Store & store) {
         StringSet newOutputNames;
         for (auto & outputName : input.second) {
             auto actualPathOpt = inputDrvOutputs.at(outputName);
-            if (!actualPathOpt)
+            if (!actualPathOpt) {
+                warn("Input %s!%s missing, aborting the resolving",
+                    store.printStorePath(input.first),
+                    outputName
```

Comment:
```suggestion
                warn("output %s of input %s missing, aborting the resolving",
                    outputName
                    store.printStorePath(input.first),
```

We probably shouldn't expose the `!output` syntax in the UI.

Reply:
Indeed (I mean we might eventually… but even if we do this might not be the proper syntax to expose)